### PR TITLE
fix(select): ensure re-opening the select resets scroll position the top

### LINKED ISSUE #6399: Scroll position reset to second item in list
### Current behaviour

Selecting drop down > scrolling to bottom of list > clicking off the drop down so it closes > reselect the drop down > the scroll has reset back to the second item in the list.

### Expected behaviour

Reset to the first item in the list. 

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/select-filterable--default

### JIRA Ticket (Sage Only)

_No response_

### Suggested Solution

_No response_

### Carbon Version

120.1.0

### Design Tokens Version

_No response_

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by robinzigmond**
Updates the react-virtual dependency, which was directly responsible for the bug. The new version makes the virtualiser return 0 items whenever the select is closed - this change in behaviour resulted in some problems after the update, particularly with tests, which have been addressed. An accessibility error with long virtualised lists, where selecting one option and scrolling away from it so that the selected option is no longer in the DOM, resulting in an invalid aria-activedescendant attribute, has also been addressed.

fix #6399

### Proposed behaviour

Whenever opening a select, the top item in the list should be visible.

No critical accessibility errors should be flagged by Axe on any Select components.

### Current behaviour

When opening a select, scrolling to the bottom, closing and then reopening, the top item cannot be seen (but can be brought into view by scrolling up).

In some circumstances (see below codesandbox and instructions to reproduce) it is possible for the select input to have an `aria-activedescendant` attribute that doesn't correspond to anything in the DOM, causing Axe to flag a critical error.

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Test that both the linked Github issue, and the accessibility issue in the below sandbox, are fixed.

Perform regression testing on all 3 types of Select, as well as the Pager component which uses a Select internally.

The following CodeSandbox shows the accessibility error with virtualised lists https://s5nzqg.csb.app/

To reproduce:
- open the select and select "option 1"
- close the select, reopen it and scroll all the way to the bottom
- with the list still open, run Axe

You should see a critical issue "ARIA attributes must conform to valid values" (see screenshot - note that the other error is a known false positive) 
image

### CODE REVIEW

File: src/components/select/select-list/select-list.component.tsx [hunk 1]
```diff
@@ -236,35 +265,39 @@ const SelectList = React.forwardRef(
...
     };
 
-    const renderedChildren = items.map((item) => {
-      const { index, start } = item;
-      const child = childrenList[index];
+    // the rangeExtractor above can cause an undefined value to be appended to the return items.
+    // Easiest way to stop that crashing is just to filter it out.
+    const renderedChildren = items
+      .filter((item) => item !== undefined)
+      .map((item) => {
```
Comment (edleeks87) [inline src/components/select/select-list/select-list.component.tsx:272]: suggestion (non-blocking): as we're not using `item` anywhere else we could destructure `index` and `start` in the map callback parameter

```suggestion
      .map(({index, start}) => {
```

File: src/components/select/select-list/select-list.spec.tsx [hunk 2]
```diff
@@ -1067,6 +1068,13 @@ describe("SelectList", () => {
       expect(optionChildren.length).toBeGreaterThanOrEqual(5);
       expect(optionChildren.length).toBeLessThan(10);
     });
+
+    it("when an option is selected, it is always in the DOM even when out of view", () => {
```
Comment (edleeks87) [inline src/components/select/select-list/select-list.spec.tsx:1072]: question (minor): is it worth having a check on visibility here as well?

File: src/components/select/filterable-select/filterable-select.spec.tsx [hunk 3]
```diff
@@ -40,6 +43,45 @@ function renderSelect(props = {}, renderer = mount) {
   return renderer(getSelect(props));
 }
 
+function simulateSelectTextboxEvent(
```
[the remark this answers is not in the document]
Comment (robinzigmond) [reply src/components/select/filterable-select/filterable-select.spec.tsx:46]: Fair point - I of course noticed the repetition but didn't really think seriously about moving them. Not sure if `spec_helper` is appropriate as they're very select-specific, but that's probably better than creating a new test helper file specifically within select. Happy to do as you suggested but will wait for comments from a second reviewer 👍

File: src/components/select/select-list/select-list.spec.tsx (hunk 2 above)
Comment (robinzigmond) [reply src/components/select/select-list/select-list.spec.tsx:1072]: is that possible in a Jest/Enzyme test? If this were a cypress/Playwright test I'd agree that asserting that it isn't visible would be useful - but it isn't (and I'm testing this differently in Cypress, by just running the accessibility checks).

Comment (edleeks87) [reply src/components/select/select-list/select-list.spec.tsx:1072]: No problem it's "nice" rather then "need" to have imo

**Review by edleeks87 — APPROVED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**